doc/spec/debug-ros-nodes.md

```diff
@@ -0,0 +1,269 @@
+        {
+            "name": "ROS: Attach to Python",
+            "type": "ros",
+            "request": "attach",
+            "runtime": "Python"
+        }
+    ]
+}
+```
+
+### Note on attach-debug
+
+Microsoft's VS Code extensions for [C++][ms-vscode.cpptools] and [Python][ms-python.python] provide flexible mechanisms of attaching into a [C++][vscode_cpp_debug] or [Python][vscode_python_debug] process.
+This extension only aims to enable a basic and generic attach-debug flow.
+Please use language-specific extensions for any specific debugging requirements including:
+
+* specifying `symbolSearchPath` or `miDebuggerPath` for C++ debugging
+* attaching to and debugging a remote process
+
+## Launch
+
+Our goal for launch-debug is to mimic the experience of debugging a single program in VS Code.
+When a `.launch` file is launched in debug mode, we want to launch all the ROS nodes defined in it in debug mode at the same time.
+Similar to attach-debug, we want to utilize all the language-specific debugging functionalities provided by `vscode-python` and `vscode-cpptools`.
+
+![launch_debug][launch_debug]
+
+From another perspective, we aim to translate launch configurations in a `.launch` file into debug configurations for VS Code that will be executed all at the same time.
+To do this, we use [the command line functionality of `roslaunch`][roslaunch_commandline] to extract launch requests from a `.launch` file:
+
+1. get parameters defined in the `.launch` file with `roslaunch --dump-params <launch-file>`
+2. load parameters with [`rosparam load`][rosparam]
+3. get the nodes defined in the `.launch` files with `roslaunch --nodes <launch-file>`
+4. get launch requests for each node with `roslaunch --args <node> <launch-file>`
+
+### TODO
+
+* [#122](https://github.com/ms-iot/vscode-ros/issues/122) support language-specific configurations in debug configuration
+
+### Limitations
+
+* when using the `--args` flag to get launch requests from the `.launch` file, environment configurations would be expressed as inline env-set commands:
+
+    ```bash
+    someenv=somevalue node arg:=value
+    ```
+
+    which would require extra (reverse) parsing of the command line.
+    This could be avoided if `roslaunch` gets [updated](#what-about-an-updated-roslaunch).
+
+* with the current implementation, all nodes in a `.launch` file, instead of [only some specified ones](#what-about-adding-a-debug-attribute-to-the-node-element-in-roslaunch), will be launched in debug mode.
+    Potentially, when too many debug sessions are launched at the same time, it would be painful to terminate them one by one.
+    To work around this, use more than 1 `.launch` files and use one of them specifically for a limited number of nodes that need to be debugged.
+
+* currently, when attempting to start multiple Python-launch debug sessions at once, only 1 of them could be connected.
+    For this reason, it is limited to have only 1 Python node in the launch file for now.
```

Comment:
I recall you mentioned this limitation is already under Python extension team's radar. It would be great we can link the respective GitHub issue here too.

Reply:
adding links to issues in vscode-python